auth/src/main/java/org/aerogear/auth/AuthService.java

```diff
@@ -0,0 +1,66 @@
+package org.aerogear.auth;
+
+import org.aerogear.auth.credentials.ICredential;
+import org.aerogear.auth.impl.AuthenticatorFactory;
+
+import java.security.Principal;
+
+/**
+ * Entry point for authenticating users.
+ */
+public class AuthService {
+
+    /**
+     * Authentication service singleton.
+     */
+    private static AuthService INSTANCE;
+
+    /**
+     * Authentication service config.
+     */
+    private final AuthServiceConfig config;
+
+    /**
+     * Instantiates a new AuthService object
+     * @param config Authentication Service configuration
+     */
+    private AuthService(final AuthServiceConfig config) {
+        this.config = config;
+    }
+
+    /**
+     * Log in the user with the given credential.
+     *
+     * @param username The username
+     * @param credentials the credential
+     * @return a user principal
+     * @throws AuthenticationException throws if the authentication fails
+     */
+    public Principal login(final String username, final ICredential credentials) throws AuthenticationException {
```

Comment:
The way it is done with OCID, we will not have the username/credential here. The login function will invoke a browser, and user will enter their username/credential in the browser. The app itself will have no access to username/credential at all for security reasons.